cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Problem.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2024, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.graph
+
+/** A problem node documenting that something failed. */
+data class Problem(
```

Comment:
@oxisto: @KuechA questions the need for the new class. I created it, because we cannot instantiate a `ProblemNode` as it is an interface. What's your preferred solution?

Reply:
Can’t we just use the existing problem expression and declarations? I would assume that we can categorize our problems into the two existing classes. Having another „Problem“ class sounds confusing to me.

Reply:
Having the additional problems property is a good idea though.

Reply:
Ok, so we would then have a `ProblemExpression` if there's a problem with a statement? Works for me - I just want to make sure that I understand you correctly.

Reply:
Since expression derives from statement, yes.

Reply:
https://github.com/Fraunhofer-AISEC/cpg/pull/1663/commits/af5e3d0f3c9fb26ae95b25ae91d92e22670468f0